### PR TITLE
test: fix autorequire luatest module

### PR DESCRIPTION
**Pull request by ochaplashkin**
### Description

`luatest` module is already available on the remote server instance. Therefore, these changes were made in many tests:

```lua
-- some_test.lua
...
local t = require('luatest')
local g = t.group()

server:exec(function()
    local t = require('luatest') -- unnecessary
    t.assert(...)
end)
```

### How the test code has been changed

All changes were made **automatically.**

#### Classic case
```lua
-- before
server:exec(function()
    local t = require('luatest')
    local do = something()
end)

-- after
server:exec(function()
    local do = something()
end)
```

#### Case with comments

```lua
-- before
exec(function()
    local t = require('luatest')

    -- Comment after require module
    local do = something()
end)

-- after
exec(function()
    -- Comment after require module
    local do = something()
end)
```

#### Group variable case

The variable group `g` was shifted down after all imports:

```lua
--before
local t = require('luatest')
local s = require('something')
local g = t.group()

--after
local t = require('luatest')
local s = require('something')

local g = t.group()
```

---

#### Some grep results before fixing

```bash
$ grep -r ":exec(" --include='*_test.lua' . | wc -l
1280
$ grep -r ":eval(" --include='*_test.lua' . | wc -l
205
$ grep -r "        local t = require('luatest')" --include='*_test.lua' .  | wc -l
655
```

#### Grep results after fixing

```bash
$ grep -r "        local t = require('luatest')" --include='*_test.lua' .  | wc -l
0
```

Part of tarantool/luatest#233

### CODE REVIEW

File: test/app-luatest/fiber_parent_backtrace_test.lua [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
 
 local g = t.group()
 
 g.before_all(function()
     g.dflt = server:new({alias = 'dflt'})
     g.dflt:start()
     g.dflt:exec(function()
-        local t = require('luatest')
         local tarantool = require('tarantool')
-
```
[the remark this answers is not in the document]
Comment (ochaplashkin) [reply test/app-luatest/fiber_parent_backtrace_test.lua:12]: Fixed.

File: test/app-luatest/interval_test.lua [hunk 2]
```diff
@@ -1,8 +1,8 @@
-local t = require('luatest')
-local g = t.group()
 local msgpack = require('msgpack')
 local msgpackffi = require('msgpackffi')
+
 local itv = require('datetime').interval
```
[the remark this answers is not in the document]
Comment (ochaplashkin) [reply test/app-luatest/interval_test.lua:3]: Added.

File: test/engine-luatest/gh_6436_field_constraint_test.lua [hunk 3]
```diff
@@ -357,7 +352,6 @@ g.test_field_constraint_nullable = function(cg)
     end)
 
     cg.server:exec(function(engine)
-        local t = require('luatest')
```
[the remark this answers is not in the document]
Comment (ochaplashkin) [reply test/engine-luatest/gh_6436_field_constraint_test.lua:360]: Deleted.

File: test/replication-luatest/gh_5295_split_brain_test.lua [hunk 4]
```diff
@@ -56,7 +53,7 @@ g.before_each(function(cg)
     -- Check that the servers start synced and with main being leader.
     -- It's a prerequisite for each test.
     cg.main:exec(function()
-        require('luatest').assert_equals(box.info.synchro.queue.owner,
+        t.assert_equals(box.info.synchro.queue.owner,
```
[the remark this answers is not in the document]
Comment (ochaplashkin) [reply test/replication-luatest/gh_5295_split_brain_test.lua:56]: Fixed.

File: test/box-luatest/gh_7217_repeatable_{in, up}sert_memtx_tx_conflict_test.lua [hunk 5]
```diff
@@ -74,8 +71,7 @@ end
 
 g.test_repeatable_upsert_primary_idx = function()
    g.server:exec(function()
-       local t = require('luatest')
-       local txn_proxy = require('test.box.lua.txn_proxy')
+              local txn_proxy = require('test.box.lua.txn_proxy')
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/box-luatest/gh_7217_repeatable_{in, up}sert_memtx_tx_conflict_test.lua]: Looks fixed.

File: test/box-luatest/gh_7217_repeatable_{in, up}sert_memtx_tx_conflict_test.lua [hunk 6]
```diff
@@ -99,8 +95,7 @@ end
...
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/box-luatest/gh_7217_repeatable_{in, up}sert_memtx_tx_conflict_test.lua]: Looks fixed.

File: test/engine-luatest/gh_6436_field_constraint_test.lua [hunk 7]
```diff
@@ -634,8 +620,7 @@ g.test_field_sql_netbox_access = function(cg)
 
     local function check_access()
         cg.server:exec(function()
-            local t = require('luatest')
-            local s = box.space.test
+                local s = box.space.test
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/engine-luatest/gh_6436_field_constraint_test.lua]: Looks fixed.

File: test/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua [hunk 8]
```diff
@@ -1,5 +1,4 @@
 local server = require('luatest.server')
-local t = require('luatest')
 local g = t.group()
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua]: Looks fixed.

File: test/box-luatest/gh_7202_options_in_atomic_test.lua [hunk 9]
```diff
@@ -20,57 +19,55 @@ end)
...
     g.server:exec(function()
-        local lt = require('luatest')
-
         -- Simple function
         local function f(...)
             return box.internal.txn_isolation(), ...
         end
         -- Simple callable table
-        local t = setmetatable({'table'}, {__call = f})
+        local table = setmetatable({'table'}, {__call = f})
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/box-luatest/gh_7202_options_in_atomic_test.lua]: Looks fixed.

File: test/replication-luatest/gh_5295_split_brain_test.lua [hunk 10]
```diff
@@ -216,7 +209,7 @@ local function perform_rollback(server)
     t.assert_gt(server:exec(function() return box.info.synchro.queue.len end), 0)
     server:exec(function() box.cfg{replication_synchro_timeout = 0.01} end)
     t.helpers.retrying({delay = 0.1}, server.exec, server, function()
-        require('luatest').assert_equals(box.info.synchro.queue.len, 0,
+        t.assert_equals(box.info.synchro.queue.len, 0,
                                          'Rollback happened')
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/replication-luatest/gh_5295_split_brain_test.lua]: Looks fixed.

File: test/replication-luatest/gh_6966_readonly_bootstrap_test.lua [hunk 11]
```diff
@@ -44,14 +42,14 @@ g.test_ro_bootstrap = function(cg)
                  'Replica receives the ER_READONLY error')
     end)
     cg.master:exec(function()
-        require('luatest').assert(box.space._cluster:count() == 1,
+        t.assert(box.space._cluster:count() == 1,
                                   'No join while master is read-only')
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/replication-luatest/gh_6966_readonly_bootstrap_test.lua]: Looks fixed.

File: test/replication-luatest/gh_6966_readonly_bootstrap_test.lua [hunk 12]
```diff
@@ -44,14 +42,14 @@ g.test_ro_bootstrap = function(cg)
...
     end)
     cg.master:eval('box.cfg{read_only = false}')
     cg.replica:wait_until_ready()
     t.helpers.retrying({}, function()
         cg.master:exec(function()
-            require('luatest').assert(box.space._cluster:count() == 2,
+            t.assert(box.space._cluster:count() == 2,
                                       'Join after master became writeable')
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/replication-luatest/gh_6966_readonly_bootstrap_test.lua]: Looks fixed.

**Review by ylobankov — APPROVED** [review]
LGTM

File: test/box-luatest/gh_7202_options_in_atomic_test.lua [hunk 13]
```diff
@@ -1,11 +1,10 @@
 local server = require('luatest.server')
-local t = require('luatest')
 
 local g = t.group()
 
 g.before_all(function()
     g.server = server:new{
-        alias  = 'default',
+        alias  = 'defaut',
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/box-luatest/gh_7202_options_in_atomic_test.lua]: Looks fixed.

File: test/replication-luatest/gh_5295_split_brain_test.lua [hunk 14]
```diff
@@ -108,16 +105,14 @@ local function reconnect_and_check_split_brain(srv)
...
         t.assert_equals(upstream.status, 'stopped', 'replication is stopped')
         t.assert_str_contains(upstream.message, 'Split-Brain discovered: ',
                               false, 'split-brain is discovered')
     end)
 end
 
 local function write_promote()
-    local t = require('luatest')
-    t.assert_not_equals(box.info.synchro.queue.owner,  box.info.id,
+        t.assert_not_equals(box.info.synchro.queue.owner,  box.info.id,
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/replication-luatest/gh_5295_split_brain_test.lua]: Looks fixed.

File: test/replication-luatest/gh_6842_qsync_applier_order_test.lua [hunk 15]
```diff
@@ -70,7 +69,7 @@ end)
 -- Wait until the server sees synchro queue owner as the given ID.
 --
 local function wait_synchro_owner(server, owner_id)
-    luatest.helpers.retrying({timeout = wait_timeout}, server.exec, server,
+    t.helpers.retrying({timeout = wait_timeout}, server.exec, server,
                              function(id)
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/replication-luatest/gh_6842_qsync_applier_order_test.lua]: Looks fixed.

File: test/replication-luatest/gh_6842_qsync_applier_order_test.lua [hunk 16]
```diff
@@ -82,7 +81,7 @@ end
 -- Wait until the server sees synchro queue has the given length.
 --
 local function wait_synchro_len(server, len_arg)
-    luatest.helpers.retrying({timeout = wait_timeout}, server.exec, server,
+    t.helpers.retrying({timeout = wait_timeout}, server.exec, server,
                              function(len)
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/replication-luatest/gh_6842_qsync_applier_order_test.lua]: Looks fixed.

File: test/replication-luatest/gh_6842_qsync_applier_order_test.lua [hunk 17]
```diff
@@ -94,7 +93,7 @@ end
 -- Wait until the server sees synchro queue as busy.
 --
 local function wait_synchro_is_busy(server)
-    luatest.helpers.retrying({timeout = wait_timeout}, server.exec, server,
+    t.helpers.retrying({timeout = wait_timeout}, server.exec, server,
                              function()
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/replication-luatest/gh_6842_qsync_applier_order_test.lua]: Looks fixed.

File: test/replication-luatest/gh_6842_qsync_applier_order_test.lua [hunk 18]
```diff
@@ -107,7 +106,7 @@ end
 -- replica ID.
 --
 local function wait_upstream_status(server, id_arg, status_arg)
-    luatest.helpers.retrying({timeout = wait_timeout}, server.exec, server,
+    t.helpers.retrying({timeout = wait_timeout}, server.exec, server,
                              function(id, status)
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/replication-luatest/gh_6842_qsync_applier_order_test.lua]: Looks fixed.

File: test/replication-luatest/gh_6842_qsync_applier_order_test.lua [hunk 19]
```diff
@@ -163,10 +162,10 @@ g.test_local_txn_during_remote_promote = function(g)
         local ok2, err2 = f2:join()
         return ok1, err1, ok2, err2
     end)
-    luatest.assert(not ok1 and not ok2 and err1 and err2,
+    t.assert(not ok1 and not ok2 and err1 and err2,
                    'both transactions failed')
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/replication-luatest/gh_6842_qsync_applier_order_test.lua]: Looks fixed.

File: test/replication-luatest/gh_6842_qsync_applier_order_test.lua [hunk 20]
```diff
@@ -163,10 +162,10 @@ g.test_local_txn_during_remote_promote = function(g)
...
-    luatest.assert_equals(err1.code, err2.code, 'same error')
-    luatest.assert_equals(err1.code, box.error.READONLY,
+    t.assert_equals(err1.code, err2.code, 'same error')
+    t.assert_equals(err1.code, box.error.READONLY,
                           'error is read-only')
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/replication-luatest/gh_6842_qsync_applier_order_test.lua]: Looks fixed.

File: test/app-luatest/uri_test.lua [hunk 21]
```diff
@@ -1,5 +1,5 @@
 local uri = require("uri")
-local t = require("luatest")
+local t = require('luatest')
```
Comment (ochaplashkin) [inline test/app-luatest/uri_test.lua:2]: Bad quotes.

File: test/box-luatest/begin_options_validation_test.lua [hunk 22]
```diff
@@ -1,5 +1,5 @@
 local server = require("luatest.server")
-local t = require("luatest")
+local t = require('luatest')
```
Comment (ochaplashkin) [inline test/box-luatest/begin_options_validation_test.lua:2]: Bad quotes too.

**Review by ochaplashkin — COMMENTED** [review]
Manual self-review completed.  2 comments were left.

File: test/app-luatest/uri_test.lua (hunk 21 above)
Comment (ylobankov) [reply test/app-luatest/uri_test.lua]: Fixed.

File: test/box-luatest/begin_options_validation_test.lua (hunk 22 above)
Comment (ylobankov) [reply test/box-luatest/begin_options_validation_test.lua]: Fixed.

File: test/replication-luatest/gh_5295_split_brain_test.lua [hunk 23]
```diff
@@ -128,8 +123,7 @@ local function write_promote()
 end
 
 local function write_demote()
-    local t = require('luatest')
-    t.assert_equals(box.info.synchro.queue.owner, box.info.id,
+        t.assert_equals(box.info.synchro.queue.owner, box.info.id,
```
[the remark this answers is not in the document]
Comment (ylobankov) [reply test/replication-luatest/gh_5295_split_brain_test.lua]: Looks like fixed.

**Comment by ylobankov** [comment]
Cherry picked to 2.10 and 1.10.

**Pull request merged**